src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -271,6 +290,46 @@ public override SyntaxNode VisitArgument(ArgumentSyntax node)
                 return newArgument;
             }
 
+            // PROTOTYPE(tuple-names) this can be memoized
```

Comment:
i wouldn't even allocate/memoize here.  tuples are going to be small enough that you can check if the name you care about shows up in the tuple exactly once quite cheaply.